backend/src/zimfarm_backend/db/requested_task.py

```diff
@@ -82,22 +87,59 @@ def request_task(
 
     # If the worker has a requested task for this schedule, return None
     if count_from_stmt(session, query) > 0:
-        return None
+        return RequestTaskResult(
+            requested_task=None, error=f"Schedule '{schedule_name}' already requested"
+        )
 
     schedule = get_schedule_or_none(session, schedule_name=schedule_name)
 
     if schedule is None or not schedule.enabled:
-        return None
+        return RequestTaskResult(
+            requested_task=None,
+            error=f"Schedule '{schedule_name}' not found or disabled",
+        )
 
     worker = None
     if worker_name is not None:
         worker = get_worker_or_none(session, worker_name=worker_name)
         if worker is None:
-            return None
+            return RequestTaskResult(
+                requested_task=None, error=f"Worker '{worker_name}' not found"
+            )
 
     # Check if worker is allowed to be assigned a task from this recipe
     if worker and schedule.context and schedule.context not in worker.contexts:
-        return None
+        return RequestTaskResult(
+            requested_task=None,
+            error=(
+                "Worker does not have required context to run schedule "
+                f"'{schedule_name}'."
+            ),
+        )
+
+    # check if worker can run the scraper for the offliner
+    if worker and schedule.config["offliner"]["offliner_id"] not in worker.offliners:
+        return RequestTaskResult(
+            requested_task=None,
+            error=(
+                f"Worker's offliners do not match the offliner for schedule "
+                f"'{schedule_name}'."
+            ),
+        )
+
+    # Check if worker has enough resource to run a task from this recipe
+    if worker and (
+        worker.cpu < schedule.config["resources"]["cpu"]
+        or worker.memory < schedule.config["resources"]["memory"]
+        or worker.disk < schedule.config["resources"]["disk"]
+    ):
+        return RequestTaskResult(
+            requested_task=None,
+            error=(
+                "Worker does not have enough resources to run schedule "
```

Comment:
Maybe list resources required by the schedule vs the ones advertised by the worker so that it is easier for the user to understand the problem?